File: hashtable/204.go
func countPrimes(n int) int {
	notPrime := make(map[int]bool, n)
    var ret int
	for i := 2; i < n; i++ {
		if !notPrime[i] {
			ret++
			for j := 2; i*j < n; j++ {
				notPrime[i*j] = true
			}
		}
	}    
    return ret
}

File: hashtable/204.cc
class Solution {
public:
    int countPrimes(int n) {
        std::vector<int> table(n, 0);
        int ret = 0;
        for (int i = 2; i < n; ++i) {
            if (table[i] == 0) {
                ++ret;
                for (int j = 2; i*j < n; ++j) {
                    table[i*j] = 1;
                }                
            }
        }
        return ret;
    }
};
